### PR TITLE
Buffing the Cellcharge and Celldrain large artifact effects

### PR DESCRIPTION
**Pull request by DeityLink**
So as it turns out, the Cell Charge and Cell Drain artifacts are both utter crap.
When the effect works as an "aura" (that means it activates every process()) it charges 25 (or drains 50) from the cell of nearby objects or silicons.
Just so you know, MoMMi spend 8 charge per Life() just by existing, and cyborgs 11.
So Cyborgs get effectively 14 charge per Life(), which with their 7.500 capacity cell would take around 20 minutes (if we account for the lag) for the large artifact to fully charge a cyborg to capacity, which takes under a minute with a Cyrborg Recharge station. A bit underwhelming for something that takes so long (and so much effort) to acquire, isn't it?

But it gets worse.

If it activates by "touch" or by "pulse", the silicon gets anywhere between 1 to 100 charge per touch (up to once per second still though) which is unreliable and mind numbing. If it activates by pulse it gets even worse still, as the borg then gets the same random amount they get by touch...except pulse can take anywhere from a few seconds to TWO MINUTES to occur, in which case the artifact just wouldn't be able to charge the borg at all.

But guess what, it gets EVEN WORSE.

The artifact also supposedly affects SMES (and anything that has a cell in it really), and it gives it just as much charge as it would give a cyborg...except an SMES has a 6.000.000 charge capacity...25 or even 100 charge are completely insignificant. Even a single solar panel could probably produce more power.

<hr>

**Therefore** in this PR, I am increasing the amount of aura charge/drain to 200 for silicons and 2.000 for SMES (which means a bit over 1% charge per minute for SMES. if you think that's low, just imagine how insignificant it was until now)
The silicon touch will give (or drain) 500 charge at once to the silicon that touch it, which is about what a borg or MoMMi gets in 2 process() spent at a recharge station.
The pulse however, will now give (or drain) 300 charge multiplied by the amount of seconds it takes for the pulse to trigger. Which means anywhere from 300 charge per second, to 36.000 charge every two minutes (oh boy that would be !fun!)

Anyway those values had never been tweaked in the past 7 years. That's my take on how to make those effects more interesting, and arguably more useful to the crew/traitors.

Also as a side effect, it now properly uses the give() and use() procs to charge or drain cells, which means that it'll explode rigged cells.

:cl:
* tweak: The cell charging and draining large artifact effects are a lot more potent now, both on silicons and on SMES (as opposed to barely having any effect like they do currently)
* bugfix: Fixed admin-spawned large artifacts being invisible by default.

### CODE REVIEW

**Comment by DrSnips** [comment]
Personally I'd rather it just be a consistent high (or low if draining) value rather than extra RNG on top of all the effort to get it.
If you threw this to robotics and told them that it's just mediocre and that the existing cell charger on the desk does it's job but twice as good with no risk why would they give a fuck?
Same for engineering for the most part. By the time you drag one of these back the power is likely set up and pumping out numbers beyond this.
Go big or go home with this.

**Comment by SonixApache** [comment]
why not just make it a percentage of the max charge?

**Comment by DeityLink** [comment]
@DrSnips if this wasn't clear, the PR does all that you suggest. I removed the RNG on touch and pulse effects, so now the amount charged/drained is reliable.

@SonixApache well recharge stations being linear as well, I thought I'd keep some consistency between the two

**Comment by DeityLink** [comment]
Fixed admin-spawned large artifacts being invisible by default.
Also fixed drain not working if it would use more than the cell's current charge.

**Comment by DrSnips** [comment]
Pardon me then the OP was a bit dense. However the charge amounts are still underwhelming imo. The rate is what normal chargers give `in two processes' meaning it's still slower than default level chargers by about half. There's no real reason for it to exist especially at the time you would find one. Just because these numbers are bigger than the last one doesn't mean it's any more viable.

**Comment by DeityLink** [comment]
@DrSnips 
>The rate is what normal chargers give `in two processes' meaning it's still slower than default level chargers by about half.

The point is that it does so often in a very huge area (up to station-wide). That means the silicons can go about their business without ever having to bother going back to a recharge station.

**Comment by DeityLink** [comment]
Also wait, you misunderstood, I meant that it gives in one process() what rechargers give in two process(), so it's twice as effective!

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -3710,6 +3710,8 @@
 							custom.secondary_effect.trigger = new custom_secondary_trigger(custom.secondary_effect)
 						custom.investigation_log(I_ARTIFACT, "|| admin-spawned by [key_name_admin(usr)] with a secondary effect [custom.secondary_effect.artifact_id]: [custom.secondary_effect] || range: [custom.secondary_effect.effectrange] || charge time: [custom.secondary_effect.chargelevelmax] || trigger: [custom.secondary_effect.trigger].")
 
+					numsuffix = 12
+					icon_state = "ano120"
```
Comment (Exxion) [inline code/modules/admin/topic.dm]:
```
code/modules/admin/topic.dm:3742:error: numsuffix: undefined var
code/modules/admin/topic.dm:3743:error: icon_state: undefined var
```

I do not know why it says those line numbers

**Comment by DeityLink** [comment]
Fixed

**Pull request merged**